internal/functions/Compare-DbaStringCollation.ps1

```diff
@@ -0,0 +1,130 @@
+    .EXAMPLE
+        PS C:\> Compare-DbaStringCollation -Reference dbá,Tools -Difference Dba,tools -Comparison In -Collation SQL_Latin1_General_CP1_CI_AI
+        Returns
+        Collation  : SQL_Latin1_General_CP1_CI_AI
+        Reference  : dbá
+        Difference : {Dba, tools}
+        Comparison : In
+        Equivalent : True
+
+        Collation  : SQL_Latin1_General_CP1_CI_AI
+        Reference  : Tools
+        Difference : {Dba, tools}
+        Comparison : In
+        Equivalent : True
+    .EXAMPLE
+        PS C:\> Compare-DbaStringCollation -Reference dbá,Tools -Difference Dba,tools -Comparison Notin -Collation SQL_Latin1_General_CP1_CI_AS
+
+        Collation  : SQL_Latin1_General_CP1_CI_AS
+        Reference  : dbá
+        Difference : {Dba, Tools}
+        Comparison : Notin
+        Result     : True
+
+        Collation  : SQL_Latin1_General_CP1_CI_AS
+        Reference  : tools
+        Difference : {Dba, Tools}
+        Comparison : Notin
+        Result     : False
+    #>
+    param(
+        [Parameter(Mandatory)]
+        [string[]]$Reference,
+        [Parameter(Mandatory)]
+        $Difference,
+        [Parameter(Mandatory)]
+        [string[]]$Collation,
+        [Parameter(Mandatory)]
+        [ValidateSet('In', 'Notin', 'Eq', 'Ne')]
+        [string]$Comparison)
+    $smo = New-Object -TypeName Microsoft.SqlServer.Management.Smo.Server
+
+    $results = @()
+    foreach ($col in $Collation) {
+        foreach ($ref in $Reference) {
+            switch ($Comparison) {
+                'Eq' { $res = $smo.getStringComparer($col).Compare($ref, $Difference) -eq 0; break; }
+                'Ne' { $res = $smo.getStringComparer($col).Compare($ref, $Difference) -ne 0; break; }
+                'In' {
+                    foreach ($dif in $Difference) {
+                        $res = $null
+                        if ($smo.getStringComparer($col).Compare($ref, $dif) -eq 0 ) {
+                            $res = $true
+                            break
+                        }
+                    }
+                    if ( $null -eq $res) { $res = $false }
+                }
+                'Notin' {
+                    foreach ($dif in $Difference) {
+                        $res = $null
+                        $testCondition = $smo.getStringComparer($col).Compare($ref, $dif)
+                        if ($testCondition -eq 0 ) {
+                            $res = $false
+                            break
+                        }
+                    }
+                    if ($null -eq $res ) { $res = $true; break; }
+                }
+            }
+            $results += [pscustomobject]@{Collation = $col
+                Reference                           = $ref
+                Difference                          = $Difference
+                Comparison                          = $Comparison
+                Result                              = $res
+            }
```

Comment:
Previous version of the code you just return the list of names. Is there any reason to have it return a full object?

Reply:
The function should be 100% contained and not require additional filtering outside of the call...unless I'm completely misunderstanding what you are trying to do.

The use of this against filtering the `$inputObject` in the primary function would be to use the output of the `Compare` function against the Database name. The only thing I'd see the compare function needing to output is an string list of the Names found matching based on the collation/string configuration. Is that not what we are trying to accomplish?

Reply:
Yes sorry for not being clear

that was meant for the  Compare-DbaStringCollation

Happy to eliminate the rest of the object, I think I understand  the no need for breaks in the switch bit now too. I was still focused on the way I originally did it. 

we only care about True for in and notin, no need to to check for false in those.

Reply:
```suggestion
```